public/src/map.js

```diff
@@ -147,6 +148,17 @@ export async function initMap(apiKey) {
     geocoder.on('result', function (e) {
       mapbox.getSource('single-point').setData(e.result.geometry);
     });
+
+    mapbox.on('click', 'constituency-highlighted', function (e) {
+      const constArea = e.features[0].properties.st_areashape;
+      if (constArea > 500000000) {
```

Comment:
Also implemented a zoom based on size of constituency so that big areas would not be zoomed too big and vice versa. Tried to multiply st_areashape with some const but seem to crash Mapbox. Hence a conditional.